dotcom-rendering/src/components/AdSlot.web.tsx

```diff
@@ -30,6 +30,7 @@ type DefaultProps = {
 	display?: ArticleDisplay;
 	isPaidContent?: boolean;
 	hasPageskin?: boolean;
+	hasShowcaseMainElement?: boolean;
```

Comment:
What's the difference between this and `display === ArticleDisplay.Showcase || display === ArticleDisplay.NumberedList`?

Reply:
If we add a new display in the future, say `ArticleDisplay.LetteredList`, it isn't obvious that it should be included in that condition and could be missed off. I also considered whether in the future we could make a change to allow `ArticleDisplay.NumberedList` to not always have a showcase main image. Therefore I thought it safer to use `article.pageType.hasShowcaseMainElement`.

Reply:
Fair point. The reason I bring this up is that part of the reason `ArticleFormat` was created was minimise the number of checks for numerous different CAPI properties all over the codebase, as we found these become hard to reason about over time. The idea is that you parse `ArticleFormat` up front, and that can be used to derive the layout of your article from then on. With this in mind, the `hasShowcaseMainElement` field looks like it fits into the former pattern rather than the latter. In addition, looking at the codebase it doesn't appear to be used anywhere else as far as I can tell, so without this usage we could be in a position to remove it?

Given this context, do you think it would be feasible to attempt to use `ArticleDisplay` for this? In this situation, to address your two points above.

> If we add a new display in the future, say ArticleDisplay.LetteredList, it isn't obvious that it should be included in that condition and could be missed off.

We could do this:

```ts
const spaceStyles = (display: ArticleDisplay): SerializedStyles => {
  switch (display) {
    case ArticleDisplay.Showcase:
    case ArticleDisplay.NumberedList:
      return css`min-height: ${doubleMpuHeight + labelHeight}px;`;
    case ArticleDisplay.Standard:
    case ArticleDisplay.Immersive:
      return css``;
  }
}
```

which means any new `ArticleDisplay`s added would cause a compilation error that would require this code to be updated[^1].

> in the future we could make a change to allow ArticleDisplay.NumberedList to not always have a showcase main image.

I agree that this scenario wouldn't be covered, but I'd also argue that it can be difficult to predict and pre-empt future design changes? This might happen, or it might never happen. It might also be the case that the main media becomes immersive, or a different aspect ratio, or something else, in which case this code might not be impacted, or might have to change in a different way?

[^1]: https://github.com/guardian/typescript-school/blob/main/4-narrowing/slides.md#exhaustiveness-checking

Reply:
OK that makes sense. I'm happy to not use `hasShowcaseMainElement` and instead use the ArticleDisplay for simplicity purposes. I will include a comment next to these styles as it's seems unlikely to an unfamiliar dev that NumberedList has a showcase main media.

I know that commercial use `hasShowcaseMainElement` on the window for Spacefinder, so this would need to be considered if we were to remove it from frontend.

I wonder about the history of NumberedList being an `ArticleDisplay`. Wouldn't it make more sense as an `ArticleFormat`? It would drastically reduce the number of combinations of `ArticleFormat` and seems similar to the `Timeline` design we worked on recently.

Reply:
> I know that commercial use `hasShowcaseMainElement` on the window for Spacefinder, so this would need to be considered if we were to remove it from frontend.

Interesting, perhaps we should discuss this?

> I wonder about the history of NumberedList being an `ArticleDisplay`. Wouldn't it make more sense as an `ArticleFormat`? It would drastically reduce the number of combinations of `ArticleFormat` and seems similar to the `Timeline` design we worked on recently.

I'm assuming you mean `ArticleDesign` here 🙂. The problem is that numbered list is often applied to articles that are represented by something else in `ArticleDesign` already, like reviews or features. In this way it's similar to immersive or showcase, which is why it's an `ArticleDisplay`. We can't make it an `ArticleDesign` because you can't have a piece that's two different values of `ArticleDesign` at the same time!